index.js

```diff
@@ -751,6 +826,30 @@
            keys.every(function(k) { return equals(self[k], other[k]); });
   }
 
+  function unionKeys(a, b) {
+    var keys = Object.keys(a);
+    var bKeys = Object.keys(b);
+    for (var i = 0; i < bKeys.length; i += 1) {
+      if (a[bKeys[i]] === void 0) keys.push(bKeys[i]);
```

Comment:
This `=== undefined` check looks suspect to me. Let's assume `a` is `{}` and `b` is `{toString: () => 'x'}`. This means `keys` will be `[]` initially and `bKeys` will be `['toString']`. `a[bKeys[0]]` will not evaluate to `undefined` because `a['toString']` evaluates to `Object.prototype.toString`.

I think the check should be `!Object.prototype.hasOwnProperty.call(a, bKeys[i])`.

This approach might be clearer:

```javascript
var key;
var keys = {};
for (key in a) keys[key] = 0;
for (key in b) keys[key] = 0;
return Object.keys(keys);
```